src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -22,7 +23,7 @@ public static partial class XmlSerializerTests
 
     static XmlSerializerTests()
     {
-        var method = typeof(XmlSerializer).GetMethod(SerializationModeSetterName);
+        var method = typeof(XmlSerializer).GetMethod(SerializationModeSetterName, BindingFlags.NonPublic | BindingFlags.Static);
```

Comment:
I've changed all `TypeInfo.GetDeclaredMethod` back to `Type.GetMethod`. For fixing this particular issue,  #16839, the change from `Type.GetMethod` to `TypeInfo.GetDeclaredMethod` was not necessary.